server/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java

```diff
@@ -103,7 +105,10 @@ static boolean shouldLoadRandomAccessFiltersEagerly(IndexSettings settings) {
         boolean loadFiltersEagerlySetting = settings.getValue(INDEX_LOAD_RANDOM_ACCESS_FILTERS_EAGERLY_SETTING);
         boolean isStateless = DiscoveryNode.isStateless(settings.getNodeSettings());
         if (isStateless) {
-            return loadFiltersEagerlySetting && DiscoveryNode.hasRole(settings.getNodeSettings(), DiscoveryNodeRole.SEARCH_ROLE);
+            return loadFiltersEagerlySetting
```

Comment:
I was not sure whether to revert this part to its original state, as it was wrong. [This](https://github.com/elastic/elasticsearch/pull/113478/files#r1785950838) has the discussion. Basically the cache should be loaded whenever the shard can be searched. But welcome feedback.

Reply:
LGTM